### PR TITLE
Change alloc sys-calls' signatures

### PR DESCRIPTION
**Pull request by ark0f**
We change signatures to be more friendly to languages other than Rust

TODO:
- [x] Blocked by https://github.com/gear-tech/dlmalloc-rust/pull/8
- [x] Blocked by #2200
- [x] OOM handler

### CODE REVIEW

File: gsys/src/lib.rs [hunk 1]
```diff
@@ -339,6 +339,18 @@ extern "C" {
     /// - `len`: `u32` length of the payload buffer.
     pub fn gr_debug(payload: *const BufferStart, len: Length);
 
+    /// Infallible `gr_panic` info syscall.
```
Comment (breathx) [inline gsys/src/lib.rs:342]:
```suggestion
    /// Infallible `gr_panic` control syscall.
```

File: gsys/src/lib.rs [hunk 2]
```diff
@@ -339,6 +339,18 @@ extern "C" {
...
+    ///
+    /// Stops the execution.
+    ///
+    /// Arguments type:
+    /// - `payload`: `const ptr` for the begging of the payload buffer.
+    /// - `len`: `u32` length of the payload buffer.
+    pub fn gr_panic(payload: *const BufferStart, len: Length) -> !;
+
+    /// Infallible `gr_oom_panic` info syscall.
```
Comment (breathx) [inline gsys/src/lib.rs:351]:
```suggestion
    /// Infallible `gr_oom_panic` control syscall.
```

File: gsys/src/lib.rs (hunk 1 above)
Comment (breathx) [reply gsys/src/lib.rs:342]: like execution-flow-control

**Review by breathx — APPROVED** [review]

**Review by grishasobol — APPROVED** [review]

File: core-backend/sandbox/src/memory.rs [hunk 3]
```diff
@@ -33,11 +33,11 @@ impl MemoryWrap {
 
 /// Memory interface for the allocator.
 impl Memory for MemoryWrap {
-    fn grow(&mut self, pages: WasmPage) -> Result<(), MemoryError> {
+    fn grow(&mut self, pages: WasmPage) -> Result<(), AllocError> {
         self.0
             .grow(pages.raw())
             .map(|_| ())
-            .map_err(|_| MemoryError::ProgramAllocOutOfBounds)
```
[the remark this answers is not in the document]
Comment (breathx) [reply core-backend/sandbox/src/memory.rs:40]: https://github.com/gear-tech/gear/issues/2257

File: core-backend/wasmi/src/memory.rs [hunk 4]
```diff
@@ -79,11 +79,11 @@ impl<E: Ext + 'static> MemoryWrap<E> {
 
 /// Memory interface for the allocator.
 impl<E: Ext + 'static> Memory for MemoryWrap<E> {
-    fn grow(&mut self, pages: WasmPage) -> Result<(), MemoryError> {
+    fn grow(&mut self, pages: WasmPage) -> Result<(), AllocError> {
         self.memory
             .grow(&mut self.store, Pages(pages.raw() as usize))
             .map(|_| ())
-            .map_err(|_| MemoryError::ProgramAllocOutOfBounds)
```
[the remark this answers is not in the document]
Comment (breathx) [reply core-backend/wasmi/src/memory.rs:86]: https://github.com/gear-tech/gear/issues/2257

**Pull request merged**